packages/eslint-plugin/src/rules/lines-around-comment.ts

```diff
@@ -0,0 +1,528 @@
+        isCommentAtParentEnd(token, AST_NODE_TYPES.SwitchStatement)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the class start or not.
+     */
+    function isCommentAtClassStart(token: TSESTree.Token): boolean {
+      return isCommentAtParentStart(token, AST_NODE_TYPES.ClassBody);
+    }
+
+    /**
+     * Returns whether or not comments are at the class end or not.
+     */
+    function isCommentAtClassEnd(token: TSESTree.Token): boolean {
+      return isCommentAtParentEnd(token, AST_NODE_TYPES.ClassBody);
+    }
+
+    /**
+     * Returns whether or not comments are at the object start or not.
+     */
+    function isCommentAtObjectStart(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentStart(token, AST_NODE_TYPES.ObjectExpression) ||
+        isCommentAtParentStart(token, AST_NODE_TYPES.ObjectPattern)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the object end or not.
+     */
+    function isCommentAtObjectEnd(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentEnd(token, AST_NODE_TYPES.ObjectExpression) ||
+        isCommentAtParentEnd(token, AST_NODE_TYPES.ObjectPattern)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the array start or not.
+     */
+    function isCommentAtArrayStart(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentStart(token, AST_NODE_TYPES.ArrayExpression) ||
+        isCommentAtParentStart(token, AST_NODE_TYPES.ArrayPattern)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the array end or not.
+     */
+    function isCommentAtArrayEnd(token: TSESTree.Token): boolean {
+      return (
+        isCommentAtParentEnd(token, AST_NODE_TYPES.ArrayExpression) ||
+        isCommentAtParentEnd(token, AST_NODE_TYPES.ArrayPattern)
+      );
+    }
+
+    /**
+     * Returns whether or not comments are at the interface start or not.
+     */
+    function isCommentAtInterfaceStart(token: TSESTree.Token): boolean {
+      return isCommentAtParentStart(token, AST_NODE_TYPES.TSInterfaceBody);
+    }
+
+    /**
+     * Returns whether or not comments are at the interface end or not.
+     */
+    function isCommentAtInterfaceEnd(token: TSESTree.Token): boolean {
+      return isCommentAtParentEnd(token, AST_NODE_TYPES.TSInterfaceBody);
+    }
+
+    /**
+     * Returns whether or not comments are at the type start or not.
+     */
+    function isCommentAtTypeStart(token: TSESTree.Token): boolean {
+      return isCommentAtParentStart(token, AST_NODE_TYPES.TSTypeLiteral);
+    }
+
+    /**
+     * Returns whether or not comments are at the type end or not.
+     */
+    function isCommentAtTypeEnd(token: TSESTree.Token): boolean {
+      return isCommentAtParentEnd(token, AST_NODE_TYPES.TSTypeLiteral);
+    }
+
+    function checkForEmptyLine(
+      token: TSESTree.Comment,
+      opts: { before?: boolean; after?: boolean },
```

Comment:
Nit:

```suggestion
      { before, after }: { before?: boolean; after?: boolean },
```

That way the `let after ..., before ...;` won't be necessary later.